### PR TITLE
Update symfony core files to 5.x

### PR DESCRIPTION
**Pull request by cavasinf**

### CODE REVIEW

File: config/preload.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+if (file_exists(dirname(__DIR__).'/var/cache/prod/App_KernelProdContainer.preload.php')) {
```
Comment (kevinpapst) [inline config/preload.php:3]: I have read a lot about problems with the preload file under 8.0...

File: config/routes/framework.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+when@dev:
```
Comment (kevinpapst) [inline config/routes/framework.yaml:1]: Is that the new style in SF5? Not using the config/packages/dev/ directory anymore?

File: composer.json [hunk 3]
```diff
@@ -51,6 +54,11 @@
             "App\\Tests\\": "tests/"
         }
     },
+    "replace": {
+        "symfony/polyfill-ctype": "*",
+        "symfony/polyfill-iconv": "*",
+        "symfony/polyfill-php72": "*"
```
Comment (kevinpapst) [inline composer.json:60]: this can be deleted, we run php 8. it would be used only in php <= 7.1

Comment (kevinpapst) [inline composer.json:60]:
```suggestion
        "symfony/polyfill-iconv": "*"
```

**Comment by kevinpapst** [comment]
Did you use the recipes upgrade logic? Or manual work?

**Review by kevinpapst — APPROVED** [review]

File: config/preload.php (hunk 1 above)
Comment (kevinpapst) [reply config/preload.php:3]: well, doesn't matter if we do not create it

File: config/routes/framework.yaml (hunk 2 above)
Comment (cavasinf) [reply config/routes/framework.yaml:1]: Yes it is, I don't now if I like it or not.. but this is what is implemented for 5.3: https://symfony.com/blog/new-in-symfony-5-3-configure-multiple-environments-in-a-single-file

> you won't be forced to use this, but if you use Symfony Flex (which is enabled by default) some files generated automatically in your project might be using it.

**Comment by cavasinf** [comment]
> recipes upgrade logic

I don't understand what do you mean by that. 😖 

-----------------------

I'm using some of those for existing project : https://github.com/symfony/symfony/blob/5.4/UPGRADE-5.0.md
This is for 4.4 to 5.0.

And I have multiple symfony version empty project from their skeleton.
Implemented with:
```bash
composer create-project symfony/skeleton:"^5.4" symfony_5_4
cd symfony_5_4
composer require webapp
```

File: config/preload.php (hunk 1 above)
Comment (cavasinf) [reply config/preload.php:3]: And if <8.0 is not allowed 😉

**Comment by kevinpapst** [comment]
> > recipes upgrade logic
> 
> I don't understand what do you mean by that. 😖

https://symfony.com/blog/fast-smart-flex-recipe-upgrades-with-recipes-update

File: config/routes/framework.yaml (hunk 2 above)
Comment (kevinpapst) [reply config/routes/framework.yaml:1]: Cool. I can remember reading that, but my project is still 4.4 so I already forgot about it again^^

**Comment by cavasinf** [comment]
Look's like magic ! 🤩

**Pull request merged**